floats/floats_test.go

```diff
@@ -180,6 +181,25 @@ func TestArgsort(t *testing.T) {
 	}
 }
 
+func TestArgsortStable(t *testing.T) {
+	for i := 1; i <= 100; i++ {
+		data := make([]float64, i+2)
+		data[0] = 2
+		data[len(data)-1] = 2
+		for j := 1; j <= i; j++ {
+			data[j] = 1
+		}
+		idx := make([]int, len(data))
+		ArgsortStable(data, idx)
+		if !sort.Float64sAreSorted(data) {
+			t.Errorf("unexpected data sort order for case %d", i)
+		}
+		if !sort.IntsAreSorted(idx[:1]) {
```

Comment:
Sorry, my typo. This should be `sort.IntsAreSorted(idx[:i])` The first element is always sorted!

Reply:
Haha, I was wondering why the test didn't fail when I changed it to be `sort.Sort`! But I figured you know best, and it was bedtime... it fails now when I switch it, so that's the right test. 👍